cmd/operator-sdk/bundle/validate.go

```diff
@@ -15,89 +15,156 @@
+  $ docker build -t quay.io/example/test-operator:v0.1.0 .
+  $ docker push quay.io/example/test-operator:v0.1.0
 
-# Generate manifests locally.
-$ operator-sdk bundle create --generate-only
+  # Ensure the image with modified metadata and Dockerfile is valid.
+  $ operator-sdk bundle validate --tag quay.io/example/test-operator:v0.1.0
 
-# Modify the metadata and Dockerfile.
-$ cd ./deploy/olm-catalog/test-operator
-$ vim ./metadata/annotations.yaml
-$ vim ./Dockerfile
+To validate the manifests directory only:
 
-# Build and push the image using the docker CLI.
-$ docker build -t quay.io/example/test-operator:v0.1.0 .
-$ docker push quay.io/example/test-operator:v0.1.0
+  $ operator-sdk bundle validate --directory ./0.1.0 --content-only
 
-# Ensure the image with modified metadata/Dockerfile is valid.
-$ operator-sdk bundle validate quay.io/example/test-operator:v0.1.0`,
+To validate the manifests and metadata:
+
+  $ operator-sdk bundle validate --directory .
+`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 1 {
-				return errors.New("a bundle image tag is a required argument, ex. example.com/test-operator:v0.1.0")
+			if err := c.validate(); err != nil {
+				return err
 			}
-			c.imageTag = args[0]
+			return c.run()
+		},
+	}
 
-			dir, err := ioutil.TempDir("", "bundle-")
-			if err != nil {
-				log.Fatal(err)
-			}
-			defer func() {
-				if err = os.RemoveAll(dir); err != nil {
-					log.Error(err.Error())
-				}
-			}()
-			logger := log.WithFields(log.Fields{
-				"container-tool": c.imageBuilder,
-				"bundle-dir":     dir,
-			})
-			log.SetLevel(log.DebugLevel)
-			val := bundle.NewImageValidator(c.imageBuilder, logger)
-			if err = val.PullBundleImage(c.imageTag, dir); err != nil {
-				log.Fatalf("Error to unpacking image: %v", err)
-			}
+	c.addToFlagSet(cmd.Flags())
 
-			log.Info("Validating bundle image format and contents")
+	return cmd
+}
 
-			if err = val.ValidateBundleFormat(dir); err != nil {
-				log.Fatalf("Bundle format validation failed: %v", err)
-			}
-			manifestsDir := filepath.Join(dir, bundle.ManifestsDir)
-			if err = val.ValidateBundleContent(manifestsDir); err != nil {
-				log.Fatalf("Bundle content validation failed: %v", err)
+func (c bundleValidateCmd) validate() error {
+	if (c.imageTag == "") == (c.directory == "") {
+		return fmt.Errorf("either --tag or --directory must be set, but not both")
+	}
+	return nil
+}
+
+func (c *bundleValidateCmd) addToFlagSet(fs *pflag.FlagSet) {
+	fs.StringVarP(&c.directory, "directory", "d", "",
+		"The directory where bundle manifests and metadata are located or, "+
+			"if --content-only is set, the path to a bundle. Either --directory or --tag must be set but not both")
+	fs.StringVarP(&c.imageTag, "tag", "t", "",
+		"The path of a registry to pull from, image name and its tag that present the bundle image "+
+			"(e.g. quay.io/test/test-operator:latest). Either --directory or --tag must be set but not both")
```

Comment:
WDYT about just keeping the CLI the same and doing the following psuedocode?
```
if argv[0] is a path to a local directory {
    validate argv[0] as bundle directory
} else {
    validate argv[0] as image
}
```

Reply:
I was thinking we could do this with `[docker|podman] image -q <image-tag>`. I'll code it out.

Reply:
It’s probably important that the directory path use case does not require docker or podman or any other image builder to be present. I’m thinking of using this in a CI or pipeline scenario where operator-sdk is running in an unprivileged container.